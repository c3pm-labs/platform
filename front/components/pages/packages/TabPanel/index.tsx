import React from 'react';

function TabPanel(props: { children: React.ReactNode, value: number, index: number}): JSX.Element {
  const {
    children, value, index, ...other
  } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && (
        <>{children}</>
      )}
    </div>
  );
}

export default TabPanel;
