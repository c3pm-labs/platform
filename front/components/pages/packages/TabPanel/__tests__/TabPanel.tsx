import { render } from '@testing-library/react';

import TabPanel from '../index';

test('TabPanel', () => {
  const { container } = render(
    <TabPanel value={1} index={1}>
      toto
    </TabPanel>,
  );

  expect(container).toMatchSnapshot();
});
