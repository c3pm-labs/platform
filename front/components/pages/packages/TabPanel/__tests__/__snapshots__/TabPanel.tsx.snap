// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel 1`] = `
<div>
  <div
    aria-labelledby="simple-tab-1"
    id="simple-tabpanel-1"
    role="tabpanel"
  >
    toto
  </div>
</div>
`;
